File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    return "".join(map(complement, list(dna_strand)))

def complement(letter):
    if letter == "G":
        return "C"
    elif letter == "C":
        return "G"
    elif letter == "T":
        return "A"
    elif letter == "A":
        return "U"

File: ruby/rna-transcription/rna_transcription.rb
class Complement
  def self.of_dna(strand)
    return "" if strand.chars.select { |c| !["A", "G", "C", "T"].include?(c) }.any?

    strand.chars.map { |letter| Complement.inverse(letter) }.join
  end

  def self.inverse(type)
    case type
    when "G"
      "C"
    when "C"
      "G"
    when "A"
      "U"
    when "T"
      "A"
    end
  end
end

module BookKeeping
  VERSION = 4
end
